File: sample/src/main/java/com/subgarden/airbrush_project/Item.java
package com.subgarden.airbrush_project;

import android.graphics.Bitmap;

import com.subgarden.airbrush.Palette;

/**
 * @author Fredrik Larsen <f@subgarden.com>
 */
public class Item {

    final private Palette palette;
    final private Bitmap image;
    final private Bitmap blurredPlaceholder;

    public Item(Bitmap image, Bitmap blurredPlaceholder) {
        this.blurredPlaceholder = blurredPlaceholder;
        this.image = image;
        this.palette = null;
    }
    public Item(Bitmap image, Palette palette) {
        this.palette = palette;
        this.image = image;
        this.blurredPlaceholder = null;
    }

    public Palette getPalette() {
        return palette;
    }

    public Bitmap getImage() {
        return image;
    }

    public Bitmap getBlurredPlaceholder() {
        return blurredPlaceholder;
    }
}

File: sample/src/main/java/com/subgarden/airbrushProject/Item.kt
package com.subgarden.airbrushProject

import android.graphics.Bitmap

import com.subgarden.airbrush.Palette

/**
 * @author Fredrik Larsen <f@subgarden.com>
 */
class Item {

    val palette: Palette?
    val blurredPlaceholder: Bitmap?
    val image: Bitmap

    constructor(image: Bitmap, blurredPlaceholder: Bitmap) {
        this.blurredPlaceholder = blurredPlaceholder
        this.image = image
        this.palette = null
    }

    constructor(image: Bitmap, palette: Palette) {
        this.palette = palette
        this.image = image
        this.blurredPlaceholder = null
    }
}
